src/Access/RoleCache.cpp

```diff
@@ -163,11 +163,15 @@ void RoleCache::roleChanged(const UUID & role_id, const RolePtr & changed_role)
 
     std::lock_guard lock{mutex};
     auto role_from_cache = cache.get(role_id);
-    if (!role_from_cache)
-        return;
```

Comment:
The problem was here - if a cache entry has expired we can't update it in the cache, that's right, but we must call `collectEnabledRoles()` anyway, which the old code didn't do.

Reply:
Yes, sure.
BTW, don't you think that collectEnabledRoles is rather expensive?